openapi.yaml

```diff
@@ -0,0 +1,143 @@
+        - name: client_secret
+          type: string
+          in: formData
+          description: OAuth Client Secret for the application (also called "Consumer Secret" in the developer portal)
+        - name: grant_type
+          type: string
+          enum: [client_credentials]
+          in: formData
+      responses:
+        '200':
+          description: authenticated successfully
+          schema:
+            properties:
+                refresh_token_expires_in:
+                  type: string
+                  example: "0"
+                api_product_list:
+                  type: string
+                  example: "[Locations]"
+                  description: a stringified JSON list of APIs this token has access to
+                api_product_list_json:
+                  type: array
+                  description: a list of APIs this token has access to
+                  items:
+                    type: string
+                    example: Locations
+                organization_name:
+                  description: the name of the Apigee organization this token was issued by
+                  type: string
+                  enum: [oregonstateuniversity]
+                developer.email:
+                  description: the email of the developer account that requested this token
+                  type: string
+                  pattern: .*@.*
+                  example: example@example.com
+                token_type:
+                  description: the type of this token
+                  type: string
+                  enum: [BearerToken]
+                issued_at:
+                  description: the unix timestamp when this token was issued
+                  type: string
+                client_id:
+                  description: the client ID of the token requester
+                  type: string
+                access_token:
+                  description: the token issued by the server
+                  type: string
+                application_name:
+                  description: the name of the application that requested this token
+                  type: string
+                scope:
+                  type: string
+                  example: ""
+                expires_in:
+                  description: the number of seconds until this token expires
+                  type: string
+                refresh_count:
+                  type: string
+                status:
+                  type: string
+                  enum: [approved]
+        '400':
+          $ref: "#/responses/400"
+        '401':
+          $ref: "#/responses/401"
+        '500':
+          $ref: "#/responses/500"
+responses:
+  '400':
+    description: Incorrect syntax in request, or required fields missing
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  '401':
+    description: User not authorized to perform this action
+    schema:
+      $ref: '#/definitions/ErrorResult'
```

Comment:
Our oauth2 endpoint doesn't respond with JSON:API errors. Instead, they look like this:
```json
{"ErrorCode" : "invalid_request", "Error" :"Required param : grant_type"}
```
or
```json
{"ErrorCode" : "invalid_client", "Error" :"Client credentials are required"}
```
Could you update these schemas?